File: Kotlin/katas/src/main/kotlin/StringCalculator.kt
import org.w3c.dom.ranges.RangeException

class StringCalculator {
    fun add(c: String): Int {
        if (c == "") return 0;
        val integers = c.split(",").map { n -> n.toInt() }
        val negatives = integers.filter { x -> x < 0 }
        if (negatives.isNotEmpty()) throw RangeException(500, "The function doesn't work with negatives")
        return integers.reduce { acc, el -> acc + el }
    }

}

File: Js-Ts/numberOne/src/string-calculator.ts
export const add = (word:string) => {
	if(word === '') return 0;
	const arr = word.split(',').map(el => parseInt(el));
	const negatives = arr.filter(x => x < 0);
	if(negatives.length > 0) throw new RangeError('Negatives are not allowed');
	const sum = arr.reduce((acc, el) => acc + el, 0);
	return sum;
}